src/pages/iou/ReceiptSelector/index.native.js

```diff
@@ -0,0 +1,313 @@
+     */
+    const showPermissionsAlert = () => {
+        Alert.alert(
+            translate('attachmentPicker.cameraPermissionRequired'),
+            translate('attachmentPicker.expensifyDoesntHaveAccessToCamera'),
+            [
+                {
+                    text: translate('common.cancel'),
+                    style: 'cancel',
+                },
+                {
+                    text: translate('common.settings'),
+                    onPress: () => Linking.openSettings(),
+                },
+            ],
+            {cancelable: false},
+        );
+    };
+
+    /**
+     * A generic handling when we don't know the exact reason for an error
+     *
+     */
+    const showGeneralAlert = () => {
+        Alert.alert(translate('attachmentPicker.attachmentError'), translate('attachmentPicker.errorWhileSelectingAttachment'));
+    };
+
+    const showCameraAlert = () => {
+        Alert.alert(translate('receipt.cameraErrorTitle'), translate('receipt.cameraErrorMessage'));
+    };
+
+    /**
+     * Common image picker handling
+     *
+     * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+     * @returns {Promise<ImagePickerResponse>}
+     */
+    const showImagePicker = (imagePickerFunc) =>
+        new Promise((resolve, reject) => {
+            imagePickerFunc(getImagePickerOptions(CONST.ATTACHMENT_PICKER_TYPE.IMAGE), (response) => {
+                if (response.didCancel) {
+                    // When the user cancelled resolve with no attachment
+                    return resolve();
+                }
+                if (response.errorCode) {
+                    switch (response.errorCode) {
+                        case 'permission':
+                            showPermissionsAlert();
+                            return resolve();
+                        default:
+                            showGeneralAlert();
+                            break;
+                    }
+
+                    return reject(new Error(`Error during attachment selection: ${response.errorMessage}`));
+                }
+
+                return resolve(response.assets);
+            });
+        });
+
+    const takePhoto = () => {
+        if (!camera.current) {
+            showCameraAlert();
+            return;
+        }
+
+        camera.current
+            .takePhoto({
+                qualityPrioritization: 'speed',
+                flash: flash ? 'on' : 'off',
+            })
+            .then((photo) => {
+                IOU.setMoneyRequestReceipt(`file://${photo.path}`, photo.path);
+                IOU.navigateToNextPage(props.iou, reportID.current, props.report);
+            })
+            .catch(() => {
+                showCameraAlert();
+            });
+    };
+
+    Camera.getCameraPermissionStatus().then((permissionStatus) => {
+        setPermissions(permissionStatus);
+    });
```

Comment:
How is this not causing an infinite render loop?

Reply:
I'm not sure, but I can confirm it does not infinite render. Happy to adjust if there is a better way to do this.

Reply:
My best guess is that `Camera.getCameraPermissionStatus` is a wrapper for RN's `getCameraPermissionStatus` and that update might be batched to avoid re-renders.